### DISCUSSION TITLE
cy.task() with three args don't work containing {log:false}

### DISCUSSION THREAD

**Discussion by guilhermealegria** [cypress-io/cypress#28762] [Questions and Help]
In construction method with custom command and cy.task I needed to insert {log:false} but when running test the console continued showed command according to prints below:
My task is with return promise and version cypress 13.2.0 with node 18.18.0

![image](https://github.com/cypress-io/cypress/assets/9544073/e15e0fbb-4f8d-4b36-bbc0-d3daa6312d5d)
![image](https://github.com/cypress-io/cypress/assets/9544073/0d477498-de69-4bfe-add5-c64192334b8d)

**Comment by portyu9** [comment]
Hi @guilhermealegria 

`{ log: false }` **is valid as the third argument to `cy.task()`**:

```js
cy.task('myTask', payload, { log: false })
```

The key detail is what `log: false` suppresses.

It only controls whether that `cy.task()` command is displayed in the **Cypress Command Log**. It does not suppress:

- `console.log()` executed by the Node task
- terminal output produced by the task
- browser DevTools console output
- logging from the surrounding custom command
- an explicit `Cypress.log()` call

For example:

```js
on('task', {
  myTask(payload) {
    console.log(payload) // still appears in the terminal
    return doSomethingAsync(payload)
  },
})
```

Returning a Promise from the Node task is supported and is unrelated to the `log` option.

So if the output shown in your screenshots is terminal/console output, the behavior is expected: remove or conditionally disable the `console.log()` calls themselves.

If the screenshots instead show the actual **`task` row still appearing in the Cypress Command Log** despite:

```js
cy.task('...', arg, { log: false })
```

then that would be different. In that case, a minimal reproduction would be useful, because the third-argument `log` option is supported.

Please mark this answer as accepted if it helped, thank you.
